api/src/backend/api/v1/views.py

```diff
@@ -780,7 +780,18 @@ def destroy(self, request, *args, **kwargs):
         if kwargs["pk"] != str(self.request.user.id):
             raise ValidationError("Only the current user can be deleted.")
 
-        return super().destroy(request, *args, **kwargs)
+        try:
+            return super().destroy(request, *args, **kwargs)
+        except Exception:
+            raise ValidationError(
+                [
+                    {
+                        "detail": "Failed to delete the user",
+                        "status": "400",
+                        "code": "delete_failed",
+                    }
+                ]
+            )
```

Comment:
Actually we don't use the "code" key, and by default this is a 400 error. So, to follow the schema I recommend using this:
```suggestion
            raise ValidationError("Failed to delete the user")
```